ext/standard/tests/array/compact_this.phpt

```diff
@@ -28,17 +28,22 @@ var_dump(
 );
 
 ?>
---EXPECT--
+--EXPECTF--
+Warning: compact(): Undefined variable: this in %s on line %d
 array(1) {
   ["this"]=>
```

Comment:
Are these `this` warnings right?

Reply:
It seems to be related to it only dealing with `this` a couple lines below in https://github.com/php/php-src/pull/3240/files#diff-497f073aa1ab88afcb8b248fc25d2a12R2595